subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -0,0 +1,317 @@
+
+#include "mesh.h"
+#include "net.h"
+#include "subnet.h"
+#include "settings.h"
+#include "brg_cfg.h"
+
+#define LOG_LEVEL CONFIG_BT_MESH_NET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_mesh_brg_cfg);
+
+/* Bridging table state. Each item is a slist node. */
+static struct brg_tbl_row brg_tbl[CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX];
+static uint32_t brg_tbl_row_cnt;
+static bool brg_enabled;
+
+static void brg_tbl_compact(void)
+{
+	int j = 0;
+
+	for (int k = 0; k < brg_tbl_row_cnt; k++) {
+		if (brg_tbl[k].direction != 0) {
+			brg_tbl[j] = brg_tbl[k];
+			j++;
+		}
+	}
+	memset(&brg_tbl[j], 0, sizeof(brg_tbl[j]));
+	brg_tbl_row_cnt--;
+}
+
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+static int brgen_set(const char *name, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	int err;
+
+	if (len_rd == 0) {
+		brg_enabled = 0;
+		LOG_DBG("Cleared bridge enable state");
+		return 0;
+	}
+
+	err = bt_mesh_settings_set(read_cb, cb_arg, &brg_enabled, sizeof(brg_enabled));
+	if (err) {
+		LOG_ERR("Failed to set bridge enable state");
+		return err;
+	}
+
+	LOG_DBG("Restored sbren");
+
+	return 0;
+}
+
+/* Define a setting for storing enable state */
+BT_MESH_SETTINGS_DEFINE(brgen, "brgen", brgen_set);
+
+/* Set function for initializing bridging table rows from brg_tbl slist */
+static int brg_tbl_set(const char *name, size_t len_rd, settings_read_cb read_cb, void *cb_arg)
+{
+	if (len_rd == 0) {
+		memset(brg_tbl, 0, sizeof(brg_tbl));
+		brg_tbl_row_cnt = 0;
+		LOG_DBG("Cleared bridging table entries");
+		return 0;
+	}
+
+	int err = bt_mesh_settings_set(read_cb, cb_arg, &brg_tbl, sizeof(brg_tbl));
+
+	if (err) {
+		LOG_ERR("Failed to set bridging table entries");
+		return err;
+	}
+
+	LOG_DBG("Restored brg_tbl");
+
+	return 0;
+}
+
+/* Define a setting for storing briging table rows */
+BT_MESH_SETTINGS_DEFINE(brgbt, "brgbt", brg_tbl_set);
+#endif
+
+bool brg_cfg_enable_get(void)
+{
+	return brg_enabled;
+}
+
+int brg_cfg_enable_set(bool enable)
+{
+	if (brg_enabled == enable) {
+		return 0;
+	}
+
+	brg_enabled = enable;
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_BRG_PENDING);
+#endif
+	return 0;
+}
+
+void bt_mesh_brg_en_pending_store(void)
+{
+#if CONFIG_BT_SETTINGS
+	char *path = "bt/mesh/brgen";
+	int err;
+
+	if (brg_enabled) {
+		err = settings_save_one(path, &brg_enabled, sizeof(brg_enabled));
+	} else {
+		err = settings_delete(path);
+	}
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path);
+	}
+#endif
+}
+
+void bt_mesh_brg_tbl_pending_store(void)
+{
+#if CONFIG_BT_SETTINGS
+	char *path = "bt/mesh/brgbt";
+	char path_bt[20];
+	int err;
+
+	if (!brg_tbl_row_cnt) {
+		err = settings_save_one(path, &brg_tbl, brg_tbl_row_cnt * sizeof(brg_tbl[0]));
+	} else {
+		err = settings_delete(path);
+	}
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path_bt);
+	}
+#endif
+}
+
+/* Remove the entry from the bridging table that corresponds with the netkey index
```

Comment:
```suggestion
/* Remove the entry from the bridging table that corresponds with the NetKey Index
```
I see we generally use capital letters for these in the docs/apis